app/services/courses/assign_program_type_service.rb

```diff
@@ -5,13 +5,18 @@ class AssignProgramTypeService
     def execute(funding_type, course)
       case funding_type
       when 'salary'
-        course.program_type = :school_direct_salaried_training_programme
+        course.program_type = if course.provider.provider_type == 'scitt'
+                                :scitt_salaried_programme
+                              elsif course.provider.provider_type == 'university'
+                                :higher_education_salaried_programme
+                              else
+                                :school_direct_salaried_training_programme
+                              end
       when 'apprenticeship'
         course.program_type = :pg_teaching_apprenticeship
       when 'fee'
         course.program_type = calculate_fee_program(course)
       end
```

Comment:
This case statements are doing the same checks, like https://github.com/DFE-Digital/publish-teacher-training/pull/3683/files#diff-41da2132a477ac75b715425fd8d48fecfb4a5e5ec9b922f56cb86afca10d45ccR10
Is there a way to dry them up?

Reply:
I don't think there's much point because we've only duplicated twice and we will be dropping the `assign_program_type_current_cycle_service` in the next cycle.